OpenRA.Mods.Common/Traits/Render/RenderVoxels.cs

```diff
@@ -59,8 +60,18 @@ public virtual IEnumerable<IActorPreview> RenderPreview(ActorPreviewInitializer
 			var palette = init.WorldRenderer.Palette(Palette ?? PlayerPalette + ownerName);
 
 			var ifacing = init.Actor.TraitInfoOrDefault<IFacingInfo>();
-			var facing = ifacing != null ? init.Contains<FacingInit>() ? init.Get<FacingInit, int>() : ifacing.GetInitialFacing() : 0;
-			var orientation = WRot.FromFacing(facing);
+			Func<WRot> orientation = () => WRot.Zero;
+			if (ifacing != null)
```

Comment:
This block is duplicated across a few places.  Can you pull it out into a helper method?